### PR TITLE
exporters/signalfx: Configure correlation from signalfx exporter

### PR DESCRIPTION
**Pull request by jrcamp**
This puts the configuration for correlation inside the signalfx exporter
config. For it to be used at runtime the exporter must be added to the traces
pipeline.

The first commit just moves the files while the second commit has changes to code.

### CODE REVIEW

File: exporter/signalfxexporter/config_test.go [hunk 1]
```diff
@@ -92,6 +95,29 @@ func TestLoadConfig(t *testing.T) {
 			},
 		},
 		DeltaTranslationTTL: 3600,
+		Correlation: &correlation.Config{
+			HTTPClientSettings: confighttp.HTTPClientSettings{
+				Endpoint: "",
+				Timeout:  5 * time.Second,
+			},
+			AccessToken:         "",
```
Comment (tigrannajaryan) [inline exporter/signalfxexporter/config_test.go:103]: Do we need a separate access token here? Can we use the top-level one?

File: exporter/signalfxexporter/config_test.go [hunk 2]
```diff
@@ -21,15 +21,18 @@ import (
 	"testing"
 	"time"
 
+	"github.com/signalfx/signalfx-agent/pkg/apm/correlations"
```
Comment (tigrannajaryan) [inline exporter/signalfxexporter/config_test.go:24]: Can we give this an alias? `correlations` is easy to confuse with `correlation` in code later.

**Comment by tigrannajaryan** [comment]
Approach looks good to me.

File: exporter/signalfxexporter/config_test.go (hunk 1 above)
Comment (jrcamp) [reply exporter/signalfxexporter/config_test.go:103]: If it's not set we default it to the top-level value. (This isn't documented in the README.md which I can add). The reason I didn't delete access token in correlation config is we use the config struct to pass to configure the correlation tracker so I would have to add another struct to capture the access token. This also keeps symmetry with endpoint where endpoint can be configured in the correlation settings but defaults to the top-level one.

I don't feel that strongly either way so if you think it's better to just have the top-level one I can remove it.

Comment (tigrannajaryan) [reply exporter/signalfxexporter/config_test.go:103]: Up to you, but it feels superfluous (unless there is a real use case where there is a need to use a different access token for correlation requests).

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Pull request merged**